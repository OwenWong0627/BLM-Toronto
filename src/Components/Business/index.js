import React from 'react';
import PropTypes from 'prop-types';

/**
 * This component represents an array of businesses that will listed on the "List Virtual Businesses Near Me"
 * page. The length of the businesses array of objects prop passed in this component will determine
 * how many li tags will be created for the businesses to be rendered on the page.
 */
const Business = ({ businesses }) => {
   return (
      <>
         <ul data-testid="business-list" className="BusinessList">
            {businesses.map((business) => (
               <li data-testid="single-business-list" className="Business" key={business.id}>
                  <h2>{business.name}</h2>
                  <h3>{business.city}</h3>
                  <h3>{business.type}</h3>
                  <p>{business.description}</p>
                  <a data-testid="business-link" target='_blank' rel="noopener noreferrer" href={business.website}>{business.website.split("://")[1]}</a>
               </li>
            ))}
         </ul>
      </>
   )
};

/**
 * propType Documentation
 */
Business.propTypes = {
   businesses: PropTypes.arrayOf(PropTypes.object).isRequired
}

export default Business;
