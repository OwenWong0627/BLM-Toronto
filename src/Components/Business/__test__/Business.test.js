import React from 'react';
import ReactDOM from 'react-dom';
import Business from '..';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

const sampleBusinesses = [{
   "id": "83",
   "name": "SHANSHAR",
   "type": "Hair, Barbers & Beauty",
   "city": "Brampton, ON",
   "website": "https://www.shanshar.com/",
   "description": "Our main goal is to WOW you by providing you with high-quality products & services. We offer kids' clothes, dÃ©cor, beauty & hair care products. They are all fabulous for less.",
   "contactInfo": "https://www.facebook.com/shansharcanada/\r\nhttps://twitter.com/ShanSharbeauty\r\nhttps://www.instagram.com/shansharbeauty/\r\nhttps://studio.youtube.com/channel/UCsG0NBp9-1mtKBdkUe99l0g/videos/upload?filter=%5B%5D&sort=%7B%22columnType%22%3A%22date%22%2C%22sortOrder%22%3A%22DESCENDING%22%7D\r\nINFO@SHANSHAR.CA \r\n"
},
{
   "id": "84",
   "name": "REAL NATURELLE",
   "type": "Hair, Barbers & Beauty",
   "city": "Brampton, ON",
   "website": "https://realnaturelle.com/",
   "description": "Our mission is to provide our clients with natural, healthy and affordable hair & skin products that improve the overall look, feel and health of all skin and hair types.",
   "contactInfo": "1-844-REAL-323 (1-844-732-5323)\r\ninfo@realnaturelle.com\r\nhttps://www.facebook.com/realnaturelle\r\nhttps://instagram.com/realnaturelle_\r\nhttps://twitter.com/realnaturelle"
},
{
   "id": "86",
   "name": "PASSION HAIR EXTENSIONS",
   "type": "Hair, Barbers & Beauty",
   "city": "Brampton, ON",
   "website": "https://passionhair10.wixsite.com/passionhairsales",
   "description": "We are a trendsetting Online Hair Extensions Store, offering our first-rate products to shoppers from the comfort of their own homes.",
   "contactInfo": "passionhair10@gmail.com\r\n"
}
];

afterEach(cleanup);
it('Renders without crashing', () => {
   const div = document.createElement("div");
   ReactDOM.render(<Business businesses={sampleBusinesses} />, div);
});

it('Renders a business item with the correct JSX tags and content', () => {
   const { getByTestId } = render(<Business businesses={sampleBusinesses.slice(0, 1)} />);
   expect(getByTestId('business-list')).toHaveTextContent(sampleBusinesses.slice(0, 1)[0].name);
   expect(getByTestId('business-list')).toHaveClass("BusinessList", { exact: true });
   expect(getByTestId('single-business-list')).toHaveClass("Business", { exact: true });
   expect(getByTestId('single-business-list')).toContainHTML("li");
   expect(getByTestId('single-business-list')).toContainHTML("h2");
   expect(getByTestId('single-business-list')).toContainHTML("h3");
   expect(getByTestId('single-business-list')).toContainHTML("p");
   expect(getByTestId('business-link')).toHaveAttribute("href", sampleBusinesses.slice(0, 1)[0].website);
   expect(getByTestId('business-link')).toHaveTextContent(sampleBusinesses.slice(0, 1)[0].website.split("://")[1]);
});

it('matches snapshot', () => {
   const tree = renderer.create(<Business businesses={sampleBusinesses} />).toJSON();
   expect(tree).toMatchSnapshot();
});