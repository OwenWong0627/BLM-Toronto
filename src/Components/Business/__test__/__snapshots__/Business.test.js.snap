// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<ul
  className="BusinessList"
  data-testid="business-list"
>
  <li
    className="Business"
    data-testid="single-business-list"
  >
    <h2>
      SHANSHAR
    </h2>
    <h3>
      Brampton, ON
    </h3>
    <h3>
      Hair, Barbers & Beauty
    </h3>
    <p>
      Our main goal is to WOW you by providing you with high-quality products & services. We offer kids' clothes, d√©cor, beauty & hair care products. They are all fabulous for less.
    </p>
    <a
      data-testid="business-link"
      href="https://www.shanshar.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      www.shanshar.com/
    </a>
  </li>
  <li
    className="Business"
    data-testid="single-business-list"
  >
    <h2>
      REAL NATURELLE
    </h2>
    <h3>
      Brampton, ON
    </h3>
    <h3>
      Hair, Barbers & Beauty
    </h3>
    <p>
      Our mission is to provide our clients with natural, healthy and affordable hair & skin products that improve the overall look, feel and health of all skin and hair types.
    </p>
    <a
      data-testid="business-link"
      href="https://realnaturelle.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      realnaturelle.com/
    </a>
  </li>
  <li
    className="Business"
    data-testid="single-business-list"
  >
    <h2>
      PASSION HAIR EXTENSIONS
    </h2>
    <h3>
      Brampton, ON
    </h3>
    <h3>
      Hair, Barbers & Beauty
    </h3>
    <p>
      We are a trendsetting Online Hair Extensions Store, offering our first-rate products to shoppers from the comfort of their own homes.
    </p>
    <a
      data-testid="business-link"
      href="https://passionhair10.wixsite.com/passionhairsales"
      rel="noopener noreferrer"
      target="_blank"
    >
      passionhair10.wixsite.com/passionhairsales
    </a>
  </li>
</ul>
`;
